### ISSUE TITLE
[no-shadow] v4 false positive: argument name and property in type have same name

### ISSUE THREAD

**Issue by doberkofler** [typescript-eslint/typescript-eslint#2454]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JS
module.exports = {
	parser: '@typescript-eslint/parser',
	parserOptions: {
		project: './tsconfig.json'
	},
	extends: [
		'eslint:recommended',
		'plugin:@typescript-eslint/recommended',
		'plugin:@typescript-eslint/recommended-requiring-type-checking',
	],
	rules: {
		'@typescript-eslint/explicit-function-return-type': 'off',
		'@typescript-eslint/indent': ['warn', 'tab'],
		'no-shadow': 'off',
		'@typescript-eslint/no-shadow': ['error'],
	}
};
```

```TS
export function render(tableId: string, callback: (tableId: string, id: number) => void ): void {
	console.log(tableId, callback);
}
```

```JSON
{
  "compilerOptions": {
    "allowJs": true,
    "checkJs": false,
    "outDir": "./dist",
    "outFile": "./dist/index.js",
    "moduleResolution": "node",
    "target": "es5",
    "module": "system",
    "sourceMap": true,
    "baseUrl": "./"
  },
  "compileOnSave": false,
  "exclude": [
    "node_modules"
  ]
}
```

**Expected Result**

valid code

**Actual Result**

  1:52  error  'tableId' is already declared in the upper scope  @typescript-eslint/no-shadow

✖ 1 problem (1 error, 0 warnings)

**Additional Info**

no

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.1` |
| `@typescript-eslint/parser`        | `4.0.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.0` |
| `node`                             | `14.8.0` |

**Comment by glen-84** [comment]
Duplicate of #2447 (?).
